src/MizarData/BaselineAndComparisonHelper.h

```diff
@@ -30,11 +32,92 @@ struct AddressToIdAndIdToSymbol {
       sfid_to_symbols;
 };
 
-[[nodiscard]] AddressToIdAndIdToSymbol AssignSampledFunctionIds(
-    const absl::flat_hash_map<orbit_mizar_base::AbsoluteAddress, orbit_mizar_base::FunctionSymbol>&
-        baseline_address_to_symbol,
-    const absl::flat_hash_map<orbit_mizar_base::AbsoluteAddress, orbit_mizar_base::FunctionSymbol>&
-        comparison_address_to_symbol);
+// `FunctionSymbolToKey` is default-constructible and defines `Key GetKey(const FunctionSymbol&)`
+// method. `Key` must be absl-hashable and define `operator==`. The functions with `FunctionSymbols`
+// mapped to equal `Key` will be assigned the same `SFID`.
+template <typename FunctionSymbolToKey>
+class BaselineAndComparisonHelperTmpl {
+  template <typename T>
+  using Baseline = ::orbit_mizar_base::Baseline<T>;
+  template <typename T>
+  using Comparison = ::orbit_mizar_base::Comparison<T>;
+  using AbsoluteAddress = ::orbit_mizar_base::AbsoluteAddress;
+  using BaselineAndComparisonFunctionSymbols =
+      ::orbit_mizar_base::BaselineAndComparisonFunctionSymbols;
+  using FunctionSymbol = ::orbit_mizar_base::FunctionSymbol;
+  using SFID = ::orbit_mizar_base::SFID;
+
+  using Key = decltype(std::declval<FunctionSymbolToKey>().GetKey(std::declval<FunctionSymbol>()));
```

Comment:
Since Valeriy asked me for a second option on the template vs. non-template. I don't really care - if it works it works.

This line is a bit ugly and would require me to read stuff up - I think this can easliy be avoided by requiring `Key `to provide a `FromSymbol` method see below.

```
template <typename Key>
class BaselineAndComparisonHelperTmpl {
...
  Key key = Key.FromSymbol(baseline_function_symbol);
...
}
```

The bigger problem here is the lack of comments in the file - it is really hard to get what all the stuff is supposed to do. Also `BaselineAndComparisonHelperTmpl` is not a great name - the thing is a match maker for debug symbols or something like that.

Reply:
In the offline discussion it was decided to have two template parameters in `BaselineAndComparisonHelperTmpl` -- `FunctionSymbolToKey` and `Key`.